extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/graal/ClientBuilderReplacement.java

```diff
@@ -2,16 +2,29 @@
 
 import javax.ws.rs.client.ClientBuilder;
 
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.client.jaxrs.engines.URLConnectionClientEngineBuilder;
+import org.jboss.resteasy.client.jaxrs.internal.LocalResteasyProviderFactory;
 import org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl;
 
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
 
+import io.quarkus.restclient.runtime.RestClientRecorder;
+
 @TargetClass(ClientBuilder.class)
 final class ClientBuilderReplacement {
 
     @Substitute
     public static ClientBuilder newBuilder() {
-        return new ResteasyClientBuilderImpl();
+        ResteasyClientBuilder client = new ResteasyClientBuilderImpl();
+        client.providerFactory(new LocalResteasyProviderFactory(RestClientRecorder.providerFactory));
+        if (!RestClientRecorder.SSL_ENABLED) {
+            client.httpEngine(new URLConnectionClientEngineBuilder().resteasyClientBuilder(client).build());
+            client.sslContext(null);
+            client.trustStore(null);
+            client.keyStore(null, "");
+        }
+        return client;
```

Comment:
@kenfinnigan can you please re-review? I took your suggestion and moved the ITs to the "main" project, which exposed a new bug on the SSL builder path in native mode. I've included the fix on these highlighted lines which mirrors what is done in `org.jboss.resteasy.microprofile.client.RestClientBuilderImpl.build(Class<T>)`